src/coreclr/inc/pgo_formatprocessing.h

```diff
@@ -361,6 +361,42 @@ bool ReadInstrumentationSchemaWithLayout(const uint8_t *pByte, size_t cbDataMax,
     });
 }
 
+
+// Return true if schemaTable entries are a subset of the schema described by pByte, with matching entries in the same order.
+// Also updates offset of the matching entries in schemaTable to those of the pByte schema.
+//
+inline bool CheckIfPgoSchemaIsCompatibleAndSetOffsets(const uint8_t *pByte, size_t cbDataMax, ICorJitInfo::PgoInstrumentationSchema* schemaTable, size_t cSchemas)
+{
+    size_t nSchema = 0;
+    size_t nMatched = 0;
+    size_t nUnmatched = 0;    
+    size_t initialOffset = cbDataMax;
+
+    auto handler = [schemaTable, &nSchema, &nMatched, &nUnmatched](const ICorJitInfo::PgoInstrumentationSchema& schema)
+    {
+        if ((schema.InstrumentationKind != schemaTable[nMatched].InstrumentationKind)
```

Comment:
Looks like we still need to guard on `nMatched < cSchemas` before this.

Reply:
Thanks. Reworked this some more.